### PR TITLE
Remove iSummaryAuthor and ISummaryCommitter

### PR DESCRIPTION
**Pull request by ghoshkaj**
## Description

Remove ISummaryAuthor and ISummaryCommitter as they have never been used. See https://github.com/microsoft/FluidFramework/issues/10456 for more details.

## Does this introduce a breaking change?

Yes, technically, public interfaces are being deprecated but since they have never been used in practicality it _should_ not affect users.

### CODE REVIEW

File: common/lib/protocol-definitions/package.json [hunk 1]
```diff
@@ -66,6 +66,14 @@
       },
       "InterfaceDeclaration_IClientConfiguration": {
         "backCompat": false
+      },
+      "InterfaceDeclaration_ISummaryAuthor": {
+        "forwardCompat": false,
+        "backCompat": false
+      },
+      "InterfaceDeclaration_ISummaryCommitter": {
```
Comment (anthony-murphy) [inline common/lib/protocol-definitions/package.json:74]: your suppression isn't quite right either. removed things get a modified name. look at the generated file for what to suppress

File: common/lib/protocol-definitions/src/test/types/validateProtocolDefinitionsPrevious.ts [hunk 2]
```diff
@@ -1074,26 +1074,26 @@ use_old_InterfaceDeclaration_ISummaryAttachment(
...
-declare function use_current_InterfaceDeclaration_ISummaryAuthor(
+declare function use_current_RemovedInterfaceDeclaration_ISummaryAuthor(
     use: TypeOnly<current.ISummaryAuthor>);
-use_current_InterfaceDeclaration_ISummaryAuthor(
+use_current_RemovedInterfaceDeclaration_ISummaryAuthor(
     get_old_InterfaceDeclaration_ISummaryAuthor());
 
 /*
-* Validate back compat by using current type in place of old type
+* Validate back compat by using current type in place ogit difff old type
```
[the remark this answers is not in the document]
Comment (ghoshkaj) [reply common/lib/protocol-definitions/src/test/types/validateProtocolDefinitionsPrevious.ts]: wowee this is embarassing! 😬 definitely didn't mean to check that in!

File: BREAKING.md [hunk 3]
```diff
@@ -59,6 +60,9 @@ The following deprecated methods are  now removed from sequence and merge-tree.
 ### Remove ConnectionState.Connecting
 `ConnectionState.Connecting` has been removed. Migrate all usage to `ConnectionState.CatchingUp` instead.
 
+### Remove ISummaryAuthor and ISummaryCommitter
+See [#10456](https://github.com/microsoft/FluidFramework/issues/10456). `ISummaryAuthor` and`ISummaryCommitter` have never been used they are being depracating.
```
Comment (NicholasCouri) [inline BREAKING.md:64]: > have never been used they are being depracating. [](http://example.com/codeflow?start=111&length=49)

as per tyler comment on the other PR(main) we might need to update this.

**Review by NicholasCouri — APPROVED** [review]
:shipit:

**Comment by ghoshkaj** [comment]
Putting off committing this in until next Tuesday to avoid annoying merge conflicts for people reconciling next with main.

**Comment by ghoshkaj** [comment]
@sonalivdeshpande would you please merge this in? I don't have the magical ability to yet!

File: BREAKING.md [hunk 4]
```diff
@@ -72,13 +72,17 @@ The following deprecated methods are  now removed from sequence and merge-tree.
 ### Remove ConnectionState.Connecting
 `ConnectionState.Connecting` has been removed. Migrate all usage to `ConnectionState.CatchingUp` instead.
 
+### Remove ISummaryAuthor and ISummaryCommitter
+See [#10456](https://github.com/microsoft/FluidFramework/issues/10456). `ISummaryAuthor` and`ISummaryCommitter` are deprecated and will be removed in a future release.
```
Comment (alexvy86) [inline BREAKING.md:76]: Should this say that they are removed in this version (2.0.0), not just deprecated and will be removed in the future?

Comment (ghoshkaj) [reply BREAKING.md:76]: Oh you're right. Let me make that correction in a different PR now since this one is already merged in.

Comment (ghoshkaj) [reply BREAKING.md:76]: Corrected here: https://github.com/microsoft/FluidFramework/pull/10981

**Pull request merged**